ts/environment.ts

```diff
@@ -62,11 +88,24 @@ export type Environment = {
   ): Package;
 };
 
+export const commonScript = (
+  env: Environment,
+  script: NixExpression,
+): NixExpression =>
+  joinNixStrings("\n", [
+    ...env.setup.filter((x) => x.tag === "common").map((x) => x.snippet),
+    script,
+  ]);
+
 export const sandboxScript = (
   env: Environment,
   script: NixExpression,
 ): NixExpression =>
-  unlinesNixStrings([nixStrLit`mkdir -p $out`, env.sandboxSetup, script]);
+  joinNixStrings("\n", [
+    nixStrLit`mkdir -p $out`,
+    ...env.setup.map((x) => x.snippet),
+    script,
+  ]);
```

Comment:
Here are two helpers, `commonScript` and `sandboxScript` that are used to create scripts that then run in either a build sandbox or outside of a build sandbox. Again, are there better ideas for names?